client/modules/Booking/Booking.js

```diff
@@ -17,37 +17,95 @@ import {
 } from "reactstrap";
 import * as http from "../../util/http";
 
-import style from './Booking.css'
+import DatePicker from 'react-datepicker';
+
+import moment from 'moment';
+
+import styles from './Booking.css'
+import {Registration} from '../Registration/Registration';
 
 const storage = require('../../util/persistedStorage');
+
 //Booking component class
 export class Booking extends Component {
 
+	intervalNum = 15;
+	intervalUnit = "minutes";
+
+	startTime = moment().add(2*this.intervalNum, this.intervalUnit);
+
 	constructor(props){
 		super(props);
 		this.state = {
-			email: '',
-			loggedIn:false,
-			booked:false,
-		};
-		this.handleSubmit = this.handleSubmit.bind(this);
-        this.canBeSubmitted = this.canBeSubmitted.bind(this);
+			car_id: '',
+			startDate: this.startTime,
+			endDate: this.startTime,
+			selectedCar:false,
+			loggedIn: false,
+			booked: false,
+			
+			touched: {
+				startDate: false,
+				endDate: false,
+			}, 
+		  };
+		  this.isFormInvalid = this.isFormInvalid.bind(this);
+	
 	}
+//Set up variables for Msgs
+
+labels = {
+	startDate: 'Start Time',
+	endDate: 'End Time',
+  };
 
+  errorMsgs = {
+	startDate: 'a valid date and time at least '+2*this.intervalNum+' '+this.intervalUnit+' in the future is required',
+	endDate: 'a date and time after start time is required',
+  };
+  
 	componentDidMount() {
 		if(storage.get(storage.Keys.JWT))
 		this.setState({	loggedIn:true });
+		console.log(Buffer.from(storage.get(storage.Keys.JWT)));
 	  }	
 
+	handleBlur(field) {
+		this.setState({ 
+			touched: Object.assign({}, this.state.touched, { [field]: true }),
+		});
+	}
+
+	validate() {
+		// true means invalid, so our conditions got reversed
+		const errs = {
+		  startDate: this.startTime.isAfter(this.state.startDate),
+		  endDate: (this.state.startDate===this.state.endDate||this.state.startDate.isAfter(this.state.endDate)),
```

Comment:
do we need this validation on the startDate? the date picker already has a minDate set

Reply:
the min date only works to check the date no the time unfortunately
and to set minTime I'd need to set maxTime and that messes up the datePicker's selectable time unfortunately

Reply:
ah the nuances of 3rd party libraries